admin/release.py

```diff
@@ -412,6 +417,66 @@ def publish_homebrew_recipe(homebrew_repo_url, version, source_bucket,
 
 
 @do
+def publish_vagrant_metadata(version, box_url, scratch_directory, box_name,
+    target_bucket):
+    """
+    Publish Vagrant metadata for a given version of a given box.
+
+    :param bytes version: The version of the Vagrant box to publish metadata
+        for.
+    :param bytes box_url: The URL of the Vagrant box.
+    :param FilePath scratch_directory: A directory to create Vagrant metadata
+        files in before uploading.
+    :param bytes box_name: The name of the Vagrant box to publish metadata for.
+    :param bytes target_bucket: S3 bucket to upload metadata to.
+    """
+    metadata_filename = '{box_name}.json'.format(box_name=box_name)
+    # Download recursively because there may not be a metadata file
+    yield Effect(DownloadS3KeyRecursively(
+        source_bucket=target_bucket,
+        source_prefix='vagrant',
+        target_path=scratch_directory,
+        filter_extensions=(metadata_filename,)))
+
+    metadata = {
+        "description": "clusterhq/{box_name} box.".format(box_name=box_name),
+        "name": "clusterhq/{box_name}".format(box_name=box_name),
+        "versions": [],
+    }
+
+    if len(scratch_directory.children()):
+        existing_metadata_file = scratch_directory.children()[0]
```

Comment:
This is kind of look-before-you-leap-y.  Instead of checking `children` and then calling it again to use, you could do it once and catch the `IndexError`.